### ISSUE TITLE
AttributeError: 'Q' object has no attribute 'split'

### ISSUE THREAD

**Issue by dex4er** [wq/django-natural-keys#8]
Hi. With 1.4.0 I'm getting following error:

```
AttributeError: 'Q' object has no attribute 'split'
```

Traceback:

```
  File "/Users/dexter/src/gui/project/admin.py", line 179, in changelist_view
    return super().changelist_view(request, extra_context)
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/contrib/admin/options.py", line 1542, in changelist_view
    self.list_max_show_all, self.list_editable, self,
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/contrib/admin/views/main.py", line 78, in __init__
    self.queryset = self.get_queryset(request)
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/contrib/admin/views/main.py", line 351, in get_queryset
    qs, search_use_distinct = self.model_admin.get_search_results(request, qs, self.query)
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/django/contrib/admin/options.py", line 910, in get_search_results
    queryset = queryset.filter(reduce(operator.or_, or_queries))
  File "/Users/dexter/src/gui/.virtualenv/lib/python3.6/site-packages/natural_keys/models.py", line 8, in filter
    slugs = natural_key_slug.split(
AttributeError: 'Q' object has no attribute 'split'
```

```
>>> natural_key_slug
<Q: (OR: ('id__iexact', '9'), ('name__istartswith', '9'))>
```

It was after I tried to use "Search" field in list view.

**Comment by marcosox** [comment]
The error seems located in `NaturalKeyQuerySet.filter()` at line 7. The function incorrectly assumes that its parameter is a natural key slug (so, a string) while actually it could be a `Q` object.
Adding a simple `if type(natural_key_slug) is str` seems to fix the issue. Will submit a pull request

**Comment by sheppard** [comment]
This is fixed in version 1.5.0 (thanks to #9, #10, and #11).
